### DISCUSSION TITLE
Option to create "Kiosk Friendly" Checkout page

### DISCUSSION THREAD

**Discussion by PatMulligan** [btcpayserver/btcpayserver#3840] [Ideas / Feature Requests]
What are thoughts on a kiosk-friendly checkout page that doesn't have any links leading away from the page (and potentially outside of a web application in a kiosk environment)? In this application, the payer only need scan the QR code invoice with their own device and so should not be presented with any other options leading to other pages/applications.

It would be nice if it could be toggled on/off for specific cases instead of globally affecting the checkout of the store since the application may also be available for people to reach on the web using their own devices, in which case they do want to have the typical checkout page with links. 

In my particular case I am using the [Create a new invoice](https://docs.btcpayserver.org/API/Greenfield/v1/#operation/Invoices_CreateInvoice) API call to generate an invoice. So I'm curious if it can be specified with an optional parameter. 

Alternatively, there could be a parameter passed into the URL of the invoice checkout link, e.g., appending #kiosk

**Comment by dstrukt** [comment]
Interesting idea, the opt-in override makes sense, and could arguably even see it as an encapsulated plugin as well.

Kiosk mode!

**Comment by NicolasDorier** [comment]
I wonder if this can't just be fixed with a custom css theme masking the button rather than adding a new option.
If that's the case, we just need to document it.

**Comment by PatMulligan** [comment]
@NicolasDorier would that allow for the option for it to be hidden in one environment and not the other? 

E.g., on the Kiosk I am using Vue for the front end and I am thinking I could pass a route parameter to the kiosk page URL to indicate that it's in a kiosk environment. Then, when the kiosk hits back-end to create the invoice it could say, hey send me the invoice checkout page with CSS that masks external links/buttons.

That way the users that reach the web app through either their browser and/or mobile would still have the "Open in Wallet" button available to them

**Reply by dennisreimann** [reply]
Right now we don't have that option, but we plan to rework the checkout page soon, so this is valuable feedback.

**Reply by bolatovumar** [reply]
@PatMulligan since invoice styling can be adjusted on a store-by-store basis why not create multiple stores and adjust the styling accordingly in one store but not the others? This way you can have a "kiosk" store with custom styling which does what you want and another one that uses just the default styling.

**Comment by PatMulligan** [comment]
The same web app on the kiosk is available to users from their own device. I suppose the kiosk could point to a different store one way or
another but then I'd also have things fragmented.

It's a restaurant menu app that I plan to open- source, so if other
restaurants use it and want to have a physical kiosk for customers to use in addition to accessing from their own devices,
it would get messy down the line having separate stores... extra storeIds,
webhooks, etc. It'd all get messy 😅
